### DISCUSSION TITLE
Running AI-Writer on Google Colab

### DISCUSSION THREAD

**Discussion by amdd252** [ALwrity/ALwrity#120] [General]
Hello everyone,

I’ve been working with the AI-Writer project on Google Colab, and I'm happy to report that the setup is working smoothly with full integration of the necessary APIs. The application is fully operational and allows for personalized content generation, keyword research, and more.

![ngrok](https://github.com/user-attachments/assets/89bf82ea-c261-4811-aa1c-3840a286ee8d)
![ngrk](https://github.com/user-attachments/assets/7863ce06-504d-412f-9c83-0a5519e7ad87)
![real ngrk](https://github.com/user-attachments/assets/f4ac5acd-ebcb-4af1-8c60-af8ec46af7f3)

**Comment by AJaySi** [comment]
Hello @amdd252 ,

Thank you so much sharing. This is great work/contribution.

Request you to please share the steps.
It will help a lot of people to get started very easily. Please include all details, such that non-tech audience like content creators can also follow your steps.

We are working on Alwrity nextjs app so as to provide AI editor, social media publishing & collaboration features.
Kindly, submit a PR, you may add your steps in wiki page. And point a link from main README

Thanks a lot.

**Reply by amdd252** [reply]
Thank you so much for your kind words! 
Steps to Get Started on AI-Writer with Google Colab
To make this accessible to content creators or non-tech users, here are the detailed steps to get started using AI-Writer on Google Colab:

Open the Google Colab Link:

Click on the link provided [here](https://colab.research.google.com/drive/1KHYDoJRDXChwCLa2G-_4AIHjNSG2lBWq?usp=sharing). This will open the Colab notebook where everything is already set up for you.

**Run the Cells:**
Once in Colab, you’ll see code cells, which are the steps needed to set up and run the AI-Writer tool. Start by running each cell in the notebook. You can do this by clicking on the play button (on the left of each cell) or selecting Runtime > Run all from the top menu.

**Mount Google Drive:**

For storage purposes, you can mount your Google Drive. This will help save any data or content generated by the tool. A prompt will appear asking you to allow Colab access to your Google Drive.

**Running the Streamlit App:**
Once the API keys are entered, the notebook will automatically spin up a Streamlit app, which you can use for content generation. A public URL will be provided using ngrok, which allows external access to the app.

**Comment by AJaySi** [comment]
@amdd252 
Thanks a lot, I have included the information in this wiki: https://github.com/AJaySi/AI-Writer/wiki/Steps-to-Get-Started-on-AI%E2%80%90Writer-with-Google-Colab

Note: It will be great if you can make github Pull request and also attach images in this wiki. 
This will also make you contributor for the project & great for your CV ;)

PS: I will invite you for exclusive beta testing the NEW Alwrity app soon.

Regards.
